### ISSUE TITLE
Model loading and warm-up takes longer than 60 seconds - Deployment Fails

### ISSUE THREAD

**Issue by tsterbak** [awslabs/amazon-sagemaker-examples#650]
Im running my own container with xgboost stacked on top of a tensorflow-based network. When I deploy the model the heath check fails, I guess because it takes to long to load the model from S3. If i train xgboost with less rounds, the deployment works but the predictive performance is of course bad.

**So is there a way to load models that take more than 60 seconds to load?** Or are there any other best practices around that? Maybe I got something wrong, because it seems to be a strange restriction...

I'm very thankful for help.

**Comment by JohnaAtAWS** [comment]
Hi tsterbak,

The model downloading timeout is about 1 hour. Could you share more specific error message or the endpoint arn here?
If you are concerning the leak of your information, you could create a customer support case with SageMaker. 

Best,
Johna

**Comment by tsterbak** [comment]
Hi Johna,
thanks for your answer. I'm using the standard scikit-learn byo container example. If I train a large model (pickled around 1.6 GB on disk) the health check fails. I get no other error messages. So I guess it must be related with loading the model to memory somehow. 
I tested:
- different instance types for deployment (even with more RAM than the training instance) 
- Different models, sklearn RandomForestClassifer and XGBClassifier (always large models)

But in both cases the health check always fails. 
Do you have any idea what else I could test or what could be the reason?

Best, 
Tobias

**Comment by laosunhust** [comment]
Hi Tobias,

Sagemaker engineer here. Could you provide either (1) region & timeline & timezone or (2) request ID when you encountered this error ? I understand you may not want to share your information here nor create support case, but with more information about your endpoint we should be able to retrieve more logs about the actual failure reason. 

You mentioned "1.6 GB on disk". Is that size before or after compression? Do you have any clue about the possible compression ratio? Sagemaker need to fit the uncompressed data in memory, so more memory-intensive instance type could be needed, depending on your compression ratio if any.
